src/ManagedServices/ManagedServices/Commands/NewAzureRmManagedServicesAssignment.cs

```diff
@@ -0,0 +1,102 @@
+        [Parameter(Position = 0, ParameterSetName = ByResourceIdParameterSet, Mandatory = false, HelpMessage = "The scope where the registration assignment should be created.")]
+        [Parameter(Position = 0, ParameterSetName = ByInputObjectParameterSet, Mandatory = false, HelpMessage = "The scope where the registration assignment should be created.")]
+        [ScopeCompleter]
+        public string Scope { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The registration definition identifier.")]
+        [ValidateNotNullOrEmpty]
+        public string RegistrationDefinitionName { get; set; }
+
+        [Parameter(ParameterSetName = ByResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "The fully qualified resource id of the registration definition.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceId")]
+        public string RegistrationDefinitionId { get; set; }
+
+        [Parameter(ParameterSetName = ByInputObjectParameterSet, ValueFromPipeline = true, Mandatory = true, HelpMessage = "The registration definition input object.")]
+        public PSRegistrationDefinition RegistrationDefinition { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public Guid RegistrationAssignmentId { get; set; } = default(Guid);
+
+        public override void ExecuteCmdlet()
+        {
+            string scope = this.GetDefaultScope();
+            string definitionId = this.RegistrationDefinitionId;
+
+            if (this.IsParameterBound(x => x.RegistrationDefinitionName))
+            {
+                scope = this.Scope ?? this.GetDefaultScope();
+                var subscriptionScope = scope.GetSubscriptionId().ToSubscriptionResourceId();
+
+                // registation definitions can only exist at the subscription level.
+                definitionId = $"{subscriptionScope}/providers/Microsoft.ManagedServices/registrationDefinitions/{this.RegistrationDefinitionName}";
+            }
+            else if (this.IsParameterBound(x => x.RegistrationDefinitionId))
+            {
+                definitionId = this.RegistrationDefinitionId;
+                scope = this.Scope ?? definitionId.GetSubscriptionId().ToSubscriptionResourceId();
+            }
+            else if (this.IsParameterBound(x => x.RegistrationDefinition))
+            {
+                definitionId = this.RegistrationDefinition.Id;
+                scope = this.Scope ?? definitionId.GetSubscriptionId().ToSubscriptionResourceId();
+            }
+
+            if (this.RegistrationAssignmentId == default(Guid))
+            {
+                this.RegistrationAssignmentId = Guid.NewGuid();
+            }
+
+            ConfirmAction(MyInvocation.InvocationName,
```

Comment:
This should identify the type of resource and identifying information (in this case, scope and managedservice definition)